### PR TITLE
Add models and custom methods

### PR DESCRIPTION
**Pull request by Ewoskyx**
# Rails Blog App Models & Custom Methods

In this PR I did;

* Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
* Set up associations between models.
### Created following custom methods
* method that returns the 3 most recent posts for a given user.
* method that updates the posts counter for a user.
* method which returns the 5 most recent comments for a given post.
* method that updates the comments counter for a post.
* method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/author.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Author < ApplicationRecord
+  has_many :posts, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  # A method that returns the 3 most recent posts for a given user.
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+end
```
Comment (alexander16108) [inline app/models/author.rb]: based on the requirement you ought to have a User model, not an Authors model setup. 😄

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @Ewoskyx , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/author.rb (hunk 1 above)
Comment (Ewoskyx) [reply app/models/author.rb]: Hey @alexander16108 ,
Thank you for the review. Sure, I'm going to fix that. ✋

**Review by descholar-ceo — APPROVED** [review]
Hi @Ewoskyx , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented the requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**